src/main/scala/ch/zzeekk/spark/temporalquery/TemporalQueryUtil.scala

```diff
@@ -211,6 +222,57 @@ object TemporalQueryUtil {
     temporalJoinImpl( df1, df2_extended, createKeyCondition(df1, df2, keys) and additionalJoinFilterCondition, "left" ).drop($"_defined")
   }
 
+  /**
+   * left outer join
+   */
+  private def temporalLeftAntiJoinImpl( df1:DataFrame, df2:DataFrame, joinColumns:Seq[String], additionalJoinFilterCondition:Column )
```

Comment:
Wow, interessant. Du hast eine ziemlich umfangreiche eigene Implementation für den left-anti join gemacht.
Aber könnten wir nicht temporalCleanupExtendImpl/temporalJoinImpl ähnlich zu temporalKeyLeftJoinImpl wiederverwenden.
- temporalCleanupExtendImpl füllt die Lücken auf df2 und erstellt _defined Columns
- temporalJoinImpl schneidet das auf die Gültigkeiten von df1 zusammen
- Anschliessend filter auf !$"_defined" um alle in df2 nicht vorhandenen Abschnitte zu selektieren

``
// extend df2
val df2_extended = temporalCleanupExtendImpl( df2, keys, rnkExpressions, Seq(), rnkFilter=true )
// left join df1 & df2
temporalJoinImpl( df1, df2_extended, createKeyCondition(df1, df2, keys) and additionalJoinFilterCondition, "left" )
.where(!$"_defined")
.select( df1.columns)
``